protocol/statetracker/state_tracker.go

```diff
@@ -89,6 +101,31 @@ func (st *StateTracker) newLavaBlock(latestBlock int64, hash string) {
 	}
 }
 
+// block handler for emergency mode checking
+func (st *StateTracker) oldLavaBlock(latestBlock int64) {
+	st.registrationLock.RLock()
+	defer st.registrationLock.RUnlock()
+
+	latestBlockTime := st.EventTracker.getLatestBlockTime()
+	downtimeParams := st.chainTracker.GetDowntimeParams()
+
+	delay := time.Now().UTC().Sub(latestBlockTime)
```

Comment:
i see the same code over and over for calculation virtual epochs, use a common function